### ISSUE TITLE
Deck breaks when looking at a K8S deploy manifest stage if there is an event that does not have a message

### ISSUE THREAD

**Issue by jeid64** [spinnaker/spinnaker#6111]
### Issue Summary:
Hello, we are using Spinnaker to deploy all kinds of Kubernetes manifests onto our cluster. We are deploying a horizontalpodautoscaler in a stage, and when we click on the stage, the Spinnaker UI breaks and a stacktrace is logged in the console.

```
TypeError: Cannot read property 'startsWith' of undefined
    at JobEventBasedPodNameProvider.canParsePodName (PodNameProvider.ts:46)
    at JobEventBasedPodNameProvider.getPodName (PodNameProvider.ts:37)
    at JobManifestPodLogs.canShow (JobManifestPodLogs.tsx:39)
    at JobManifestPodLogs.render (JobManifestPodLogs.tsx:82)
    at Qg (react-dom.production.min.js:173)
    at Og (react-dom.production.min.js:172)
    at Tg (react-dom.production.min.js:180)
    at bi (react-dom.production.min.js:232)
    at ci (react-dom.production.min.js:233)
    at Di (react-dom.production.min.js:249) "Possibly unhandled rejection: {}"
```

We looked into this problem a lot, and it seems to be triggered by having a ManifestEvent with no "message" field on it. If the manifest has one or more events on it,  [Deck will try to make a JobManifestPodLogs object](https://github.com/spinnaker/deck/blob/32c8cc8e2c2ad07066ff1c1345fb0d07f213f55c/app/scripts/modules/kubernetes/src/pipelines/stages/deployManifest/manifestStatus/ManifestEvents.tsx#L79)

Eventually that JobManifestPodLogs object will call [canParsePodName](https://github.com/spinnaker/deck/blob/f0287a11ccf92ab902adb55de8664bcde43175bd/app/scripts/modules/core/src/manifest/PodNameProvider.ts#L46) which then attempts to check if the "message" field on manifestEvent [startsWith "Created pod".](https://github.com/spinnaker/deck/blob/f0287a11ccf92ab902adb55de8664bcde43175bd/app/scripts/modules/core/src/manifest/PodNameProvider.ts#L46)  However, our manifestEvent does not have a "message" field
![image](https://user-images.githubusercontent.com/1271427/95778391-0a28b600-0c96-11eb-9a3b-1a6a8ff0f426.png)

so startsWith does not exist. The Kubernetes API [says that the message field is optional](https://github.com/kubernetes/api/blob/751d93c667b4ab26e388f5d1ad36e6e89622cf03/core/v1/types.go#L5296). In our case, the program creating the Event does not include a message, which is valid, but breaks Deck when Deck tries to parse the ManifestEvent.

I would probably just add a check to see if message is null in ManifestEvent before trying to access it.
https://github.com/spinnaker/deck/blob/8c01a5667c693d14a4b4c9240e0d94c94ed3e19b/app/scripts/modules/core/src/domain/IManifest.ts#L28
has it as optional, so normally the compiler would break you because the current code doesn't check for nullability, https://github.com/spinnaker/deck/blob/ede5b906e0af21a76b9845909fa38ee02db4e5d8/tsconfig.json#L61 but tsconfig is set to disable nullability checks.

### Cloud Provider(s):
Kubernetes

### Feature Area:
UI

### Description:
Behavior I expect to see:
When I click the deploy manifest stage for a pipeline, it should load the information about the stage.
Actual behavior:
![image](https://user-images.githubusercontent.com/1271427/95778901-0c3f4480-0c97-11eb-8edf-65428685c357.png)

### Steps to Reproduce:

1. Create a horizontalpodautoscaler manifest in a DeployManifest stage
2. Have something add a Kubernetes Event with an empty string for message (In our case, [it was datadog adding "" for an event it makes when it takes over ownership of an HPA](https://github.com/DataDog/datadog-agent/blob/3a147e81e3976bd3e65c707ec8913163ab1f74a2/pkg/util/kubernetes/apiserver/hpa_controller.go#L173)
3. Go to the DeployManifest stage in Deck and you'll now have a broken UI.

**Comment by maggieneterval** [comment]
Hi @jeid64! Thanks for the thorough issue report. I agree with your proposed solution of adding a nullability check to [`canParseName`](https://github.com/spinnaker/deck/blob/f0287a11ccf92ab902adb55de8664bcde43175bd/app/scripts/modules/core/src/manifest/PodNameProvider.ts#L40) to ensure the message is present before trying to parse it. Would you like to make a PR to resolve this issue?

**Comment by jeid64** [comment]
@maggieneterval Opened a PR to fix it, thanks!

**Comment by maggieneterval** [comment]
@jeid64 Awesome, much appreciated! Closing this issue, and your fix should be released with Spinnaker 1.24.
